### PR TITLE
Remove transitive dependencies from sonar-plugin-api

### PR DESCRIPTION
**Pull request by simonbrandhof**
This is another step before supporting guava 18.

### CODE REVIEW

File: sonar-duplications/src/main/java/net/sourceforge/pmd/cpd/SourceCode.java [hunk 1]
```diff
@@ -59,9 +62,7 @@
     protected abstract Reader getReader() throws Exception;
 
     protected List<String> load() {
-      LineNumberReader lnr = null;
-      try {
-        lnr = new LineNumberReader(getReader());
+      try (LineNumberReader lnr = new LineNumberReader(getReader())) {
```
Comment (julienlancelot) [inline sonar-duplications/src/main/java/net/sourceforge/pmd/cpd/SourceCode.java]: Jacoco will not be happy with that...

**Comment by julienlancelot** [comment]
On my side it's LGTM. I let @sns-seb give its own LGTM.

File: sonar-duplications/src/main/java/net/sourceforge/pmd/cpd/SourceCode.java (hunk 1 above)
Comment (simonbrandhof) [inline sonar-duplications/src/main/java/net/sourceforge/pmd/cpd/SourceCode.java]: Indeed but on purpose.

Comment (julienlancelot) [inline sonar-duplications/src/main/java/net/sourceforge/pmd/cpd/SourceCode.java]: Could you just tell me why then ?

Comment (sns-seb) [inline sonar-duplications/src/main/java/net/sourceforge/pmd/cpd/SourceCode.java]: what? Jacoco does not support try-with-resource?
That would be the reason I haven't many of them in code yet.

Comment (julienlancelot) [inline sonar-duplications/src/main/java/net/sourceforge/pmd/cpd/SourceCode.java]: Indeed, Jacoco does not currently "understand" try-with-resource, it badly detect 8 branches to cover.
See https://github.com/jacoco/jacoco/issues/15 for the whole story.

**Comment by sns-seb** [comment]
well, I gave it a high altitude eye and LGTM

**Pull request closed without merging**